### ISSUE TITLE
Truffle testing with Solidity ^0.5; DeployedAddresses not supporting payable addresses 

### ISSUE THREAD

**Issue by ryanwdavies** [trufflesuite/truffle#1640]
## Issue
In Solidity 0.5 addresses are of the type "payable" and "non-payable".

Truffle's DeployedAddresses appears to only support non-payable addresses, which does not allow it to be used with contracts which have a payable fallback function.

## Steps to Reproduce

example contract:
````javascript
pragma solidity 0.5.2;

contract ThamesBoats {
  address owner;

  constructor() public {
      owner = msg.sender;
  }

  /// @notice fallback funtion, payable
  function() external payable { }
}
````

Test.sol
````javascript
pragma solidity 0.5.2;

contract TestThamesBoats {

  ThamesBoats public thamesBoats;
  thamesBoats = ThamesBoats(DeployedAddresses.ThamesBoats());

  function () external payable {}
}
````

Running the test script yields:
````javascript
truffle test

TypeError: Explicit type conversion not allowed from non-payable "address" to "contract ThamesBoats", which has a payable fallback function.
      thamesBoats = ThamesBoats(DeployedAddresses .ThamesBoats());
                    ^------------------------------------------^
````

## Expected Behavior

If I repeat the same except use "pragma solidity 0.4.25", all is well and the contract address is returned by DeployedAddresses in the normal way.

## Actual Results

Error above.

## Environment

* Operating System: Ubuntu 18.04.1 LTS
* Ethereum client: 
* Truffle version (`truffle version`):
Truffle v5.0.0 (core: 5.0.0)
Solidity - 0.5.2 (solc-js)
Node v8.12.0

* node version (`node --version`): v8.12.0
* npm version (`npm --version`): 6.4.1

**Comment by eggplantzzz** [comment]
I think we should have a fix for this with the release of `0.5.3`. We plan on releasing today (January 30).

**Comment by eggplantzzz** [comment]
truffle@5.0.3 was just released and should solve this issue! Closing this but let me know if you still experience problems.
